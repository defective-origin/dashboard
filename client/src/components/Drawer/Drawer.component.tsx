import React, { useState } from 'react'
import { Drawer as MuiDrawer } from '@mui/material'

// ---| core |---
import { cn } from 'tools'
import { useFunc } from 'hooks'

// ---| pages |---
// ---| screens |---
// ---| components |---

// ---| self |---
import css from './Drawer.module.scss'

export type DrawerProps = {
  className?: string
  children?: React.ReactNode
  side?: 'left' | 'right' | 'top' | 'bottom'
  onClose?: () => void
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <Drawer />
 */
export function Drawer(props: DrawerProps): JSX.Element {
  const { side = 'right', children, className, ...otherProps } = props
  const _className = cn(css.Drawer, className)
  const [open, isOpen] = useState(false)
  const toggle = useFunc(() => isOpen((open) => !open))

  return (
    <MuiDrawer
      className={_className}
      anchor={side}
      open={open}
      onClose={toggle}
      onClick={toggle}
      {...otherProps}
    >
      {children}
    </MuiDrawer>
  )
}

Drawer.displayName = 'Drawer'

export default Drawer
