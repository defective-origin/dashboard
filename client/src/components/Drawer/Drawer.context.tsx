import React, { createContext, useContext } from 'react'

// ---| common |---
import { CounterReturnOptions, useCounter } from 'common/hooks'

export const DrawerContext = createContext<CounterReturnOptions | null>(null)

export const useDrawerContext = (): CounterReturnOptions | null => useContext(DrawerContext)

export const DrawerContextProvider = (props: React.PropsWithChildren): JSX.Element => {
  const asideCounter = useCounter()

  return <DrawerContext.Provider value={asideCounter} {...props} />
}
