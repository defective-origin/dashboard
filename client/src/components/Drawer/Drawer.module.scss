@import "Launcher/UILauncher/ThemeProvider/ThemeProvider.module.scss";

.Visible {
  background: $shadow-color;
  z-index: 1;
}
