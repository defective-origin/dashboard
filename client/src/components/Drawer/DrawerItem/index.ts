export { default } from './DrawerItem.component'
export * from './DrawerItem.component'
