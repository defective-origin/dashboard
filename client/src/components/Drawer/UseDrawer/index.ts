/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './UseDrawer.hook'

/* INJECT_EXPORT_PLACE */
export * from './UseDrawer.hook'
