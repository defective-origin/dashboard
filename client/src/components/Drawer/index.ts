export { default } from './Drawer.component'
export * from './Drawer.component'

export { default as DrawerItem } from './DrawerItem'
export * from './DrawerItem'
