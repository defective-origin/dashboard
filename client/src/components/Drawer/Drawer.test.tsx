// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Drawer from './Drawer.component'
// import useDrawer from './Drawer.hook'
// import DrawerProvider, Drawer from './Drawer.context'

describe('[Drawer] component', () => {
  it('should render component', () => {
    const container = render(<Drawer />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})

// describe('[Drawer] hook', () => {
//   it('should return options', () => {
//     const { result } = renderHook(() => useDrawer())
// 
//     expect(result.current).toBe(null)
//   })
// })

// describe('[Drawer] context', () => {
//   const wrapper = (props: DrawerProps) => <DrawerProvider { ...props } />
// 
//   it('should return options', () => {
//     const { result } = renderHook(() => useDrawer(), { wrapper })
// 
//     expect(result.current).toBeTruthy()
//   })
// })
