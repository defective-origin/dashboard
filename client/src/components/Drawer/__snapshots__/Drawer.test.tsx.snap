// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Drawer] component > should render component 1`] = `<DocumentFragment />`;
